src/map/atcommand.c

```diff
@@ -8945,13 +8945,17 @@ ACMD(accinfo)
 /* [Ind] */
 ACMD(set)
 {
-	char reg[SCRIPT_VARNAME_LENGTH+1], val[254];
+	char reg[SCRIPT_VARNAME_LENGTH + 1];
+	char val[SCRIPT_STRING_VAR_LENGTH + 1];
 	struct script_data* data;
 	int toset = 0;
 	bool is_str = false;
 	size_t len;
 
-	if (!*message || (toset = sscanf(message, "%32s %253[^\n]", reg, val)) < 1) {
+	char format[20];
+	snprintf(format, sizeof(format), "%%%ds %%%d[^\n]", SCRIPT_VARNAME_LENGTH, SCRIPT_STRING_VAR_LENGTH);
```

Comment:
better use safesnprintf

Reply:
also if SCRIPT_VARNAME_LENGTH will be 2^32, will be buffer overflow

Reply:
Why should someone set `SCRIPT_VARNAME_LENGTH` to 4294967296? o.O

Reply:
> better use safesnprintf
Fixed.

> also if SCRIPT_VARNAME_LENGTH will be 2^32, will be buffer overflow
Added static assert check for SCRIPT_VARNAME_LENGTH.